kusto/ingest/status.go

```diff
@@ -0,0 +1,301 @@
+	// Exhausted represents a retryable failure that has exhusted all retry attempts
+	Exhausted FailureStatusCode = "Exhausted"
+)
+
+// IsRetryable indicates whether there's any merit in retying ingestion
+func (i FailureStatusCode) IsRetryable() bool {
+	switch i {
+	case Transient:
+		return true
+
+	case Exhausted:
+		return true
+
+	default:
+		return false
+	}
+}
+
+// statusRecord is a record containing information regarding the status of an ingation command
+type statusRecord struct {
+	// Status - The ingestion status returned from the service. Status remains 'Pending' during the ingestion process and
+	// is updated by the service once the ingestion completes. When <see cref="IngestionReportMethod"/> is set to 'Queue', the ingestion status
+	// will always be 'Queued' and the caller needs to query the reports queues for ingestion status, as configured. To query statuses that were
+	// reported to queue, see: <see href="https://docs.microsoft.com/en-us/azure/kusto/api/netfx/kusto-ingest-client-status#ingestion-status-in-azure-queue"/>.
+	// When <see cref="IngestionReportMethod"/> is set to 'Table', call <see cref="IKustoIngestionResult.GetIngestionStatusBySourceId"/> or
+	// <see cref="IKustoIngestionResult.GetIngestionStatusCollection"/> to retrieve the most recent ingestion status.
+	Status StatusCode
+
+	// IngestionSourceID - A unique identifier representing the ingested source. It can be supplied during the ingestion execution.
+	IngestionSourceID uuid.UUID
+
+	// The URI of the blob, potentially including the secret needed to access
+	// the blob. This can be a filesystem URI (on-premises deployments only),
+	// or an Azure Blob Storage URI (including a SAS key or a semicolon followed
+	// by the account key).
+	IngestionSourcePath string
+
+	// Database - The name of the database holding the target table.
+	Database string
+
+	// Table - The name of the target table into which the data will be ingested.
+	Table string
+
+	// UpdatedOn - The last updated time of the ingestion status.
+	UpdatedOn time.Time
+
+	// OperationID - The ingestion's operation ID.
+	OperationID uuid.UUID
+
+	// ActivityID - The ingestion's activity ID.
+	ActivityID uuid.UUID
+
+	// ErrorCode In case of a failure, indicates the failure's error code.
+	// TODO [Yochai, July 2020] make this into a const list
+	ErrorCode string
+
+	// FailureStatus - In case of a failure, indicates the failure's status.
+	FailureStatus FailureStatusCode
+
+	// Details - In case of a failure, indicates the failure's details.
+	Details string
+
+	// OriginatesFromUpdatePolicy - In case of a failure, indicates whether or not the failure originated from an Update Policy.
+	OriginatesFromUpdatePolicy bool
+}
+
+// newStatusRecord creates a new record initialized with defaults and user provided data.
+func newStatusRecord() statusRecord {
+	rec := statusRecord{
+		Status:                     Failed,
+		IngestionSourceID:          uuid.Nil,
+		IngestionSourcePath:        "Undefined",

```

Comment:
Make "Undefined" a constant so you can use it here and below in other functions like  FromProps().  This prevents misspelling mistakes now and the future.